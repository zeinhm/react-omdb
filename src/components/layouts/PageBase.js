import React, { useContext, useEffect } from 'react';
import PropTypes from 'prop-types';
import { ModalContext } from '../../contexts';

export default function PageBase({ children }) {
  const contextModal = useContext(ModalContext);

  useEffect(() => {
    const body = document.getElementsByTagName('body')[0];
    if (contextModal?.content) {
      body.className = 'modal-open';
    }
    if(!contextModal?.content){
      body.className = '';
    }
  }, [contextModal?.content]);

  return (
    <>
      {children}
      <div className="modal">{contextModal?.content}</div>
    </>
  );
}

PageBase.defaultProps = {
  children: null,
};

PageBase.propTypes = {
  children: PropTypes.node,
};

