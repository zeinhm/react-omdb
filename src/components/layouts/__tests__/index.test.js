import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import PageBase from '../PageBase';

jest.mock('../../../contexts', () => ({
  ModalContext: {
    content: '',
    setContent: jest.fn(),
  },
  AlertContext: {
    alert: '',
    setAlert: jest.fn(),
  },
}));

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useContext: v => v,
  useEffect: cb => cb(),
}));

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(() => jest.fn()),
}));

describe('src/components/layouts/PageBase', ()=> {
  test('PageBase', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<PageBase />);
    expect(tree).toMatchSnapshot();
  });
});
