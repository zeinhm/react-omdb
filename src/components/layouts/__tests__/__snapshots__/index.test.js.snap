// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/layouts/PageBase PageBase 1`] = `
<React.Fragment>
  <div
    className="modal"
  >
    
  </div>
</React.Fragment>
`;
